libraries/stdlib/jdk8/src/kotlin/jvm/optionals/Optionals.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2010-2022 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package kotlin.jvm.optionals
+
+import java.util.Optional
+
+/**
+ * Returns this [Optional]'s value if [present][Optional.isPresent], or otherwise `null`.
+ */
+public fun <T : Any> Optional<out T>.getOrNull(): T? = orElse(null)
```

Comment:
Seemingly `out` would make this function technically more similar to something like `List.firstOrNull()`, since `List`'s type parameter is declared `out` but `Optional`'s is not, IIUC (https://youtrack.jetbrains.com/issue/KT-49210). So in the `toList()` case for instance, `out` seems to be needed to allow something like `Optional.of("foo").toList<CharSequence>()`.

But I agree it doesn't seem to have practical value here, so I removed `out` except in the `toXxx` functions, where it seems to enable some corner cases (see above), for which I added tests. Let me know if you want me to remove `out` (and the extra tests) for `toXxx` as well, though.